ftw/linkchecker/command/checking_links.py

```diff
@@ -19,31 +27,159 @@ def get_plone_sites_information(app):
     return plone_sites_information
 
 
-def setup_plone(plone_site):
+def setup_plone(app, site_info):
+    plone_site = app.get(site_info['id'])
     user = AccessControl.SecurityManagement.SpecialUsers.system
     user = user.__of__(plone_site.acl_users)
     newSecurityManager(plone_site, user)
     setSite(plone_site)
 
 
-def get_relation_catalog_for_plone_site(app, PloneSiteId):
-    plone_site = app.get(PloneSiteId)
-    setup_plone(plone_site)
-
-    return getUtility(ICatalog)
+def get_broken_relations_in_plone_page():
+    relation_catalog = getUtility(ICatalog)
+    catalog_lookup(relation_catalog)
+    # there's need to also check another catalog
```

Comment:
I'm not quite sure, whether the links, not found that way, all are in the fields searched later in code anyway (`@ get_external_broken_links_in_plone_page()`). If not we need to find another catalog containing these.